### ISSUE TITLE
Top-level constants appear in class scope

### ISSUE THREAD

**Issue by chrisseaton** [oracle/truffleruby#1443]
I found this while trying to work out where `Queue` was defined - because it also is accessible as `Thread::Queue`. It turns out *all* constants can be used like that in TruffleRuby!

```ruby
p Queue
p Thread::Queue

class Whatever
end

p Whatever
p Thread::Whatever
```

This doesn't work in MRI.

**Comment by eregon** [comment]
I believe that's correct behavior as of Ruby 2.4.4:
```
$ chruby 2.4.4             
$ ruby -e 'p String::Array'
-e:1: warning: toplevel constant Array referenced by String::Array
Array

$ chruby 2.5.0
$ ruby -e 'p String::Array'
Traceback (most recent call last):
-e:1:in `<main>': uninitialized constant String::Array (NameError)
Did you mean?  Array
```
Except that we don't produce the warning.

Note in 2.5 the toplevel lookup was removed.
